### PR TITLE
MAN-26: My cases search filter

### PR DESCRIPTION
**Pull request by pmcphee77**

### CODE REVIEW

File: assets/js/backendSortableTable.js [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+MOJFrontend.BackendSortableTable.prototype.setupOptions = function (params) {
+  params = params || {}
+  this.statusMessage = params.statusMessage || 'Sort by %heading% (%direction%)'
+  this.ascendingText = params.ascendingText || 'ascending'
+  this.descendingText = params.descendingText || 'descending'
+}
+
+MOJFrontend.BackendSortableTable.prototype.createHeadingButtons = function () {
+  var headings = this.table.find('thead th')
+  var heading
```
Comment (marcus-bcl) [inline assets/js/backendSortableTable.js:27]: These `var`s can be `const` / `let` - all browsers support it now.

File: assets/scss/application.scss [hunk 2]
```diff
@@ -11,8 +11,36 @@ $govuk-page-width: $moj-page-width;
...
+  font-size: 16px;
+}
+
+.nowrap {
+  white-space: nowrap; /* Prevents text from wrapping */
+  min-width: 110px;
+}
+
+.govuk-select--width-20 {
+  max-width: 41ex
```
Comment (marcus-bcl) [inline assets/scss/application.scss:44]: Is this supposed to be `ex`, or is it a typo of `px`?  Seems odd to use x-height for a width - but not sure if that's intentional?

File: integration_tests/e2e/yourCases.cy.ts
Comment (marcus-bcl) [inline integration_tests/e2e/yourCases.cy.ts:1]: Guess we should rename the file too?

File: integration_tests/pages/yourCases.ts
Comment (marcus-bcl) [inline integration_tests/pages/yourCases.ts:1]: Same here

File: server/data/masApiClient.ts [hunk 3]
```diff
@@ -101,6 +101,20 @@ export default class MasApiClient extends RestClient {
...
+    username: string,
+    page: string,
+    sortBy: string,
+    body: Record<never, never> = {},
+  ): Promise<UserCaseload> {
+    let pageQuery = '?size=10'
+    if (page) {
+      pageQuery = `${pageQuery}&page=${page}`
+    }
+    pageQuery = `${pageQuery}&sortBy=${sortBy}`
```
Comment (marcus-bcl) [inline server/data/masApiClient.ts:114]: Might be worth looking at [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) for this
```suggestion
    const pageQuery = new URLSearchParams({ size: "10", page, sortBy }).toString();
```

File: server/utils/nunjucksSetup.ts [hunk 4]
```diff
@@ -99,6 +100,33 @@ export default function nunjucksSetup(app: express.Express, applicationInfo: App
   njkEnv.addFilter('activityLogDate', activityLogDate)
   njkEnv.addFilter('removeEmpty', removeEmpty)
   njkEnv.addFilter('toSlug', toSlug)
+
+  app.use((req, res, next) => {
+    /* eslint-disable */
+    njkEnv.addFilter('decorateFormAttributes', (obj: any, section: string, filterName: string) => {
```
Comment (marcus-bcl) [inline server/utils/nunjucksSetup.ts:106]: Not sure I understand this... but guessing it's just copied from the MOJ sortable table component?

File: assets/scss/application.scss [hunk 5]
```diff
@@ -11,8 +11,36 @@ $govuk-page-width: $moj-page-width;
...
 @import './components/card';
 @import 'assets/scss/local';
+@import './components/sortable-table';
 @import "../../node_modules/govuk-frontend/dist/govuk/base";
 @import "../../node_modules/@ministryofjustice/frontend/moj/components/sortable-table/_sortable-table";
-@import './components/sortable-table';
 
+// TM styles
+.secondary-text {color: $govuk-secondary-text-colour; /* Using a GOV.UK Design System color */}
+.strong { font-weight: 900;}
```
Comment (marcus-bcl) [inline assets/scss/application.scss:20]: Did you try using the `govuk-!-font-weight-bold` class, or is it not bold enough?

**Review by marcus-bcl — COMMENTED** [review]
Couple of minor things - but looks good, will be cool to see it in action

File: assets/scss/application.scss (hunk 5 above)
Comment (pmcphee77) [reply assets/scss/application.scss:20]: I think Tim wanted something slightly lighter than the usual bold

File: server/utils/nunjucksSetup.ts (hunk 4 above)
Comment (pmcphee77) [reply server/utils/nunjucksSetup.ts]: Its actually taken from the original prototype. Its a filter on a form attribute that sets the value in session and defaults the form to the value held in session. Was trying to make it easier. Ive tried to change it a bit to make it more readable. Can refactor later if needed

**Review by marcus-bcl — DISMISSED** [review]

**Review by marcus-bcl — APPROVED** [review]

**Pull request merged**